users/service.go

```diff
@@ -85,15 +93,15 @@ func (svc service) RegisterClient(ctx context.Context, token string, cli mgclien
 	}()
 	client, err := svc.clients.Save(ctx, cli)
 	if err != nil {
-		return mgclients.Client{}, errors.Wrap(repoerr.ErrCreateEntity, err)
+		return mgclients.Client{}, errors.Wrap(svcerr.ErrCreateEntity, err)
 	}
 	return client, nil
 }
 
 func (svc service) IssueToken(ctx context.Context, identity, secret, domainID string) (*magistrala.Token, error) {
 	dbUser, err := svc.clients.RetrieveByIdentity(ctx, identity)
 	if err != nil {
-		return &magistrala.Token{}, errors.Wrap(svcerr.ErrViewEntity, err)
+		return &magistrala.Token{}, err
```

Comment:
Please Wrap this error with `svcerr.ErrAuthentication`, most of the times, error here  will be repoerr.ErrNotFound, So i suggest to wrap with `svcerr.ErrAuthentication`.